tests/utils/local-network-utils/local_network_setup.go

```diff
@@ -0,0 +1,402 @@
+			teleporterCode, err := subnetInfo.ChainRPCClient.CodeAt(ctx, teleporterContractAddress, nil)
+			Expect(err).Should(BeNil())
+			Expect(len(teleporterCode)).Should(BeNumerically(">", 2)) // 0x is an EOA, contract returns the bytecode
+		}
+		log.Info("Finished deploying Teleporter contract", "blockchainID", subnetInfo.BlockchainID.Hex())
+	}
+	log.Info("Deployed Teleporter contracts to all subnets")
+}
+
+func DeployTeleporterRegistryContracts(teleporterAddress common.Address) {
+	log.Info("Deploying TeleporterRegistry contract to subnets")
+	ctx := context.Background()
+
+	entries := []teleporterregistry.ProtocolRegistryEntry{
+		{
+			Version:         big.NewInt(1),
+			ProtocolAddress: teleporterAddress,
+		},
+	}
+
+	subnetAInfo := *subnetsInfo[subnetA]
+	subnetBInfo := *subnetsInfo[subnetB]
+
+	var (
+		err error
+		tx  *types.Transaction
+	)
+	optsA := utils.CreateTransactorOpts(ctx, subnetAInfo, fundedAddress, fundedKey)
+	teleporterRegistryAddressA, tx, _, err := teleporterregistry.DeployTeleporterRegistry(optsA, subnetAInfo.ChainRPCClient, entries)
+	Expect(err).Should(BeNil())
+	subnetsInfo[subnetA].TeleporterRegistryAddress = teleporterRegistryAddressA
+	// Wait for the transaction to be mined
+	receipt, err := bind.WaitMined(ctx, subnetAInfo.ChainRPCClient, tx)
+	Expect(err).Should(BeNil())
+	Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+
+	optsB := utils.CreateTransactorOpts(ctx, subnetBInfo, fundedAddress, fundedKey)
+	teleporterRegistryAddressB, tx, _, err := teleporterregistry.DeployTeleporterRegistry(optsB, subnetBInfo.ChainRPCClient, entries)
+	Expect(err).Should(BeNil())
+	subnetsInfo[subnetB].TeleporterRegistryAddress = teleporterRegistryAddressB
+
+	// Wait for the transaction to be mined
+	receipt, err = bind.WaitMined(ctx, subnetBInfo.ChainRPCClient, tx)
+	Expect(err).Should(BeNil())
+	Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+
+	log.Info("Deployed TeleporterRegistry contracts to all subnets")
+}
+
+func DeployMockToken(ctx context.Context, fundedAddress common.Address, fundedKey *ecdsa.PrivateKey, source utils.SubnetTestInfo) (common.Address, *exampleerc20.ExampleERC20) {
+	opts := utils.CreateTransactorOpts(ctx, source, fundedAddress, fundedKey)
+
+	// Deploy Mock ERC20 contract
+	address, txn, mockToken, err := exampleerc20.DeployExampleERC20(opts, source.ChainRPCClient)
+	Expect(err).Should(BeNil())
+	log.Info("Deployed Mock ERC20 contract", "address", address.Hex(), "txHash", txn.Hash().Hex())
+
+	// Wait for the transaction to be mined
+	receipt, err := bind.WaitMined(ctx, source.ChainRPCClient, txn)
+	Expect(err).Should(BeNil())
+	Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+
+	return address, mockToken
+}
+
+func DeployExampleCrossChainMessenger(ctx context.Context, fundedAddress common.Address, fundedKey *ecdsa.PrivateKey, source utils.SubnetTestInfo) (common.Address, *examplecrosschainmessenger.ExampleCrossChainMessenger) {
```

Comment:
some utils functions pass in fundedKey/address, while some others use the var set at top, should we stick with one of them throughout?

Reply:
Good call. Adjusted all functions that use these values to include them as parameters